gulp/util.js

```diff
@@ -223,3 +225,75 @@ function themeBuildStream() {
       .pipe(sass())
       .pipe(utils.cssToNgConstant('material.core', '$MD_THEME_CSS'));
 }
+
+// Removes duplicated CSS properties.
+function dedupeCss() {
+  var prefixes = ['-webkit-', '-moz-', '-ms-', '-o-'];
+
+  return insert.transform(function(contents) {
+    // Parse the CSS into an AST.
+    var parsed = postcss.parse(contents);
+    var removedSelectors = [];
+    var removedCount = 0;
+
+    // Walk through all the rules, skipping comments, media queries etc.
+    parsed.walk(function(rule) {
+      if (rule.type !== 'rule' || !rule.nodes || rule.nodes.length < 2) return;
+
+      // Walk all of the properties within a rule.
+      rule.walk(function(prop) {
+        // Check if there's a similar property that comes after the current one.
+        // Note that we skip the prefixed properties and values, because they should
+        // be allowed to be repeated.
+        var hasDuplicates = find(rule.nodes, function(otherProp) {
+          return otherProp !== prop &&
+            prop.prop &&
+            otherProp.prop &&
+            otherProp.prop === prop.prop &&
+            !containsPrefix(prop.prop) &&
+            !containsPrefix(prop.value) &&
+            !containsPrefix(otherProp.prop) &&
+            !containsPrefix(otherProp.value);
```

Comment:
Would it make sense to remove the unnecessary iteration here?

Current approach
```scss
selector {
  -webkit-transform: someValue; // Iteration 1x (iterate through all properties) -> Do nothing (prefixed)
  -ms-transform: someValue;     // Iteration 2x (iterate through all properties) -> Do nothing (prefixed)
  transform: someValue;         // Iteration 3x (iterate through all properties) -> delete
  transform: anotherValue;      // Iteration 4x (iterate through all properties) -> None found. Do nothing
}
```

New approach
```scss
selector {
  -webkit-transform: someValue; // Iteration 1x (prefixed - skip)
  -ms-transform: someValue;     // Iteration 2x (prefixed - skip)
  transform: someValue;         // Iteration 3x (iterate through all properties) -> delete
  transform: anotherValue;      // Iteration 4x (iterate through all properties) -> None found. Do nothing
}
```